table/tables/index.go

```diff
@@ -293,7 +293,11 @@ func (c *index) Create(sctx sessionctx.Context, rm kv.RetrieverMutator, indexedV
 		// If the index kv was untouched(unchanged), and the key/value already exists in mem-buffer,
 		// should not overwrite the key with un-commit flag.
 		// So if the key exists, just do nothing and return.
-		_, err = txn.GetMemBuffer().Get(ctx, key)
+		if memTxn, ok := txn.(kv.MemBufferRetriever); ok {
+			_, err = memTxn.GetFromMemBuffer(ctx, key)
```

Comment:
I think it's better to change TxnState.GetMembuffer implementation, so we will not misuse it later.

Reply:
The buffer level is too complex to get things right.

Reply:
great